cmake/onnxruntime_python.cmake

```diff
@@ -60,7 +60,7 @@ onnxruntime_add_include_to_target(onnxruntime_pybind11_state gsl)
 if(APPLE)
   set(ONNXRUNTIME_SO_LINK_FLAG "-Xlinker -exported_symbols_list ${ONNXRUNTIME_ROOT}/python/exported_symbols.lst")
 elseif(UNIX)
-  set(ONNXRUNTIME_SO_LINK_FLAG "-Xlinker --version-script=${ONNXRUNTIME_ROOT}/python/version_script.lds -Xlinker --no-undefined -Xlinker --gc-sections")
+  set(ONNXRUNTIME_SO_LINK_FLAG "-Xlinker --version-script=${ONNXRUNTIME_ROOT}/python/version_script.lds -Xlinker --gc-sections")
```

Comment:
Please revert this line back.

Reply:
I added the "--no-undefined" flag here in purpose.  If you can't build onnxruntime without the flag, then it means you missed something in the link inputs.  This flag is a safety check for us to know what's going on.  I'd rather prefer to keep it.

Reply:
@snnn, manylinux images do not provide a library to link against for python on purpose. All those symbols will be provided by the python executable (built statically). That just works the way symbol resolution is done on Linux but prevents the usage of  "--no-undefined" flag here. It was defined this way in order to allow usage of manylinux wheels wether python was built with static or dynamic libraries.